lib/dialogs/timer_dialog.dart

```diff
@@ -26,18 +52,18 @@ class TimerDialog {
       builder: (BuildContext context) {
         return StatefulBuilder(
           builder: (context, setState) {
-            // Initialize the timer only once
             if (timer == null) {
-              playSound('assets/start_sound.mp3');
+              playSound('start_sound.ogg');
               timer = Timer.periodic(const Duration(seconds: 1), (timer) {
                 if (!isPaused) {
                   if (remainingSeconds == 0) {
                     timer.cancel();
-                    playSound('assets/end_sound.mp3');
-                    Navigator.of(context).pop();
-                    if (onComplete != null) {
-                      onComplete();
-                    }
+                    playSound('end_sound.ogg', onComplete: () {
```

Comment:
The selected line:
```diff
[36,60]-                    playSound('assets/end_sound.mp3'); [<- only line selection]
```
is changed to:
```diff
[36,61]+                    playSound('end_sound.ogg', onComplete: () {
```
This change modifies the sound file format from `mp3` to `ogg` for the sound that plays when the timer ends. Additionally, it introduces an `onComplete` callback to the `playSound` function. This callback will execute additional code after the sound finishes playing, specifically to close the dialog and trigger the `onComplete` function if it's provided.